api/v1alpha4/awsmachine_webhook.go

```diff
@@ -149,6 +149,14 @@ func (r *AWSMachine) validateRootVolume() field.ErrorList {
 		allErrs = append(allErrs, field.Required(field.NewPath("spec.rootVolumeOptions.iops"), "iops required if type is 'io1' or 'io2'"))
 	}
 
+	if r.Spec.RootVolume.Type == "gp2" && r.Spec.RootVolume.Throughput != 0 {
```

Comment:
We don't only support gp2 AFAIK.
Please take a look at the webhook validation check in this PR for gp3: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2496/files

Reply:
thanks! you are correct, I forgot that gp2 support is provided by the legacy aws provider in kubernetes.